pkg/render/node.go

```diff
@@ -490,6 +536,99 @@ func (c *nodeComponent) nodeRole() *rbacv1.ClusterRole {
 	return role
 }
 
+// nodeRole creates the clusterrole containing policy rules that allow the node daemonset to operate normally.
+func (c *nodeComponent) CNIPluginRole() *rbacv1.ClusterRole {
+	finalizer := []string{}
+	if !c.cfg.Terminating {
+		finalizer = []string{NodeFinalizer}
+	}
+	role := &rbacv1.ClusterRole{
+		TypeMeta: metav1.TypeMeta{Kind: "ClusterRole", APIVersion: "rbac.authorization.k8s.io/v1"},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:       CalicoCNIPluginObjectName,
+			Labels:     map[string]string{},
+			Finalizers: finalizer,
+		},
+
+		Rules: []rbacv1.PolicyRule{
+			{
+				// The CNI plugin needs to get pods, nodes, namespaces.
+				APIGroups: []string{""},
+				Resources: []string{"pods", "nodes", "namespaces"},
+				Verbs:     []string{"get"},
+			},
+			{
+				// Calico patches the allocated IP onto the pod.
+				APIGroups: []string{""},
+				Resources: []string{"pods/status"},
+				Verbs:     []string{"patch"},
+			},
+			{
+				// Most IPAM resources need full CRUD permissions so we can allocate and
+				// release IP addresses for pods.
+				APIGroups: []string{"crd.projectcalico.org"},
+				Resources: []string{
+					"blockaffinities",
+					"ipamblocks",
+					"ipamhandles",
+					"ipamconfigs",
+				},
+				Verbs: []string{"get", "list", "create", "update", "delete"},
+			},
+			{
+				// But, we only need to be able to query for IPAM config.
+				APIGroups: []string{"crd.projectcalico.org"},
+				Resources: []string{"ipamconfigs"},
+				Verbs:     []string{"get"},
+			},
+			{
+				// Calico creates some configuration on startup.
+				APIGroups: []string{"crd.projectcalico.org"},
+				Resources: []string{
+					"clusterinformations",
+					"felixconfigurations",
+					"ippools",
+				},
+				Verbs: []string{"create", "update"},
+			},
+			{
+				// For monitoring Calico-specific configuration.
+				APIGroups: []string{"crd.projectcalico.org"},
+				Resources: []string{
+					"bgpconfigurations",
+					"bgppeers",
+					"blockaffinities",
+					"clusterinformations",
+					"felixconfigurations",
+					"globalnetworkpolicies",
+					"stagedglobalnetworkpolicies",
+					"globalnetworksets",
+					"hostendpoints",
+					"ipamblocks",
+					"ippools",
+					"ipreservations",
+					"networkpolicies",
+					"stagedkubernetesnetworkpolicies",
+					"stagednetworkpolicies",
+					"networksets",
+				},
+				Verbs: []string{"get", "list", "watch"},
```

Comment:
The CNI plugin doesn't perform any watches, so we can remove that permission. 

It's probably moot though - I also don't think it needs most of these resources, so we can probably just delete this entire block. The only ones it should need are the IPAM resources which are already covered above.